src/components/MyDevices/MyDevices.react.js

```diff
@@ -0,0 +1,109 @@
+import '../StaticAppBar/StaticAppBar.css';
+import $ from 'jquery';
+import PropTypes from 'prop-types';
+import Cookies from 'universal-cookie';
+import React, { Component } from 'react';
+import TableComplex from '../TableComplex/TableComplex.react';
+import StaticAppBar from '../StaticAppBar/StaticAppBar.react';
+import UserPreferencesStore from '../../stores/UserPreferencesStore';
+
+import './MyDevices.css'
+
+const cookies = new Cookies();
+let obj=[];
+
+let defaults = UserPreferencesStore.getPreferences();
+let defaultServerURL = defaults.Server;
+
+let BASE_URL = '';
+if(cookies.get('serverUrl')===defaultServerURL||
+  cookies.get('serverUrl')===null||
+  cookies.get('serverUrl')=== undefined) {
+  BASE_URL = defaultServerURL;
+}
+else{
+  BASE_URL= cookies.get('serverUrl');
+}
+
+let url = BASE_URL+'/aaa/listUserSettings.json?'
+        +'access_token='+cookies.get('loggedIn');
+
+// eslint-disable-next-line
+class MyDevices extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            dataFetched: false,
+        };
+    }
+
+    apiCall = () => {
+      $.ajax({
+        url: url,
+        type:'GET',
+        dataType: 'jsonp',
+        statusCode: {
+          404: function(xhr) {
+            if(window.console) {console.log(xhr.responseText);}
+            console.log('Error 404: Resources not found!');
+          },
+          503: function(xhr) {
+            if(window.console) {console.log(xhr.responseText);}
+            console.log('Error 503: Server not responding!');
+          }
+        },
+        crossDomain: true,
+        timeout: 3000,
+        async: false,
+        success: function (response) {
+            let keys = Object.keys(response.devices);
+            for (var i = 0; i < keys.length; i++) {
+
+              var myObj = {
+                  'macid' : keys[i],
+                  'devicename' : response.devices[keys[i]]
+              };
+
+              obj.push( myObj );
+
+              this.setState({
+                dataFetched: true,
+              });
+            }
+        }.bind(this),
+        error: function(errorThrown){
+          console.log(errorThrown);
+        }
+      });
+    }
+
+    // eslint-disable-next-line
+    componentWillMount(){
```

Comment:
Why not cDM?

Reply:
cDM was resulting in mounting/unmounting issues, that's why.

Reply:
Which issues exactly, I think it should work fine. :thinking: